app/workers/asset_manager_update_asset_worker.rb

```diff
@@ -3,6 +3,11 @@ class AssetManagerUpdateAssetWorker < WorkerBase
 
   def perform(legacy_url_path, new_attributes = {})
     attributes = find_asset_by(legacy_url_path)
+
+    if (replacement_path = new_attributes.delete('replacement_legacy_url_path'))
+      new_attributes['replacement_id'] = find_asset_by(replacement_path)['id']
```

Comment:
Is it safe to assume that the replacement asset will exist in Asset Manager at this point?

Reply:
@chrisroos:

The replacement `AttachmentData` will definitely have been created in Whitehall, but you're right that there's no guarantee that the corresponding asset(s) will have been created in Asset Manager by this point. However, in this scenario, I'm confident that the update job will fail, because it will get a `404 Not Found` from Asset Manager when looking up the replacement asset ID (or possibly a `422 Unprocessable Entity` if it tries to do the update with a replacement asset ID which does not exist). The job will then be retried until the replacement asset is found and the update is executed successfully. While this isn't ideal, I think it will give the desired outcome.

Do you think this is OK? Can you think of a better way of doing it? Here are a couple of ideas:

1. Only have a single dedicated Sidekiq worker processing Asset Manager jobs so that they are always executed in the order they are queued. This doesn't feel great, because it will reduce the throughput of jobs.

2. In some ways I think the only problem with the solution in this PR is that exceptions will be reported in Sentry which are not actually real problems. Perhaps we could somehow allow a small number of retries for the update job if the asset lookup fails in very specific ways _without raising an exception_...?

What do you think?

Reply:
I think that having the job fail and retry is absolutely fine - I don't think we should do anything more than that at the moment. I guess the main concern would be that the job succeeded but didn't actually set the replacement as expected.

Reply:
@chrisroos Cool. Thanks. 👍

Reply:
> I guess the main concern would be that the job succeeded but didn't actually set the replacement as expected.

I might be missing something, but I can't see how this could happen. On that basis, I'm happy to merge this PR. @chrisroos please say if you think I've missed something.